promcda/models/ProMCDA.py

```diff
@@ -0,0 +1,519 @@
+        robustness_weights = True
+        robustness_indicators = False
+        marginal_distributions = (PDFType.NORMAL, PDFType.UNIFORM, PDFType.NORMAL)
+
+        promcda = ProMCDA(input_matrix=input_matrix,
+                          polarity=polarity,
+                          robustness_weights=robustness_weights,
+                          robustness_indicators=robustness_indicators,
+                          marginal_distributions=marginal_distributions,
+                          num_runs=5000,
+                          num_cores=2,
+                          random_seed=123)
+
+        # Run normalization, aggregation, and MCDA methods
+        df_normalized = promcda.normalize()
+        df_aggregated = promcda.aggregate()
+        promcda.run_mcda()
+        """
+
+        self.input_matrix = input_matrix
+        self.polarity = polarity
+        self.weights = weights.copy() if weights is not None else None
+        self.robustness_weights = robustness_weights
+        self.robustness_single_weights = robustness_single_weights
+        self.robustness_indicators = robustness_indicators
+        self.num_runs = num_runs
+        self.marginal_distributions = marginal_distributions
+        self.num_cores = num_cores
+        self.random_seed = random_seed
+        self.normalized_values_without_robustness = None
+        self.normalized_values_with_robustness = None
+        self.aggregated_scores = None
+        self.all_indicators_scores_means = None
+        self.all_indicators_scores_stds = None
+        self.all_indicators_means_scores_normalized = None
+        self.all_indicators_scores_stds_normalized = None
+        self.all_weights_score_means = None
+        self.all_weights_score_stds = None,
+        self.all_weights_score_means_normalized = None,
+        self.all_weights_score_stds_normalized = None,
+        self.iterative_random_w_score_means = None,
+        self.iterative_random_w_score_stds = None,
+        self.iterative_random_w_score_means_normalized = None
+
+        self.input_matrix_no_alternatives = check_input_matrix(self.input_matrix)
+
+        if self.weights is None and robustness_indicators is False:
+            self.weights = [0.5] * self.input_matrix_no_alternatives.shape[1]
+        elif self.weights is None and robustness_indicators is True:
+            self.input_matrix, num_indicators, polarity, weights = process_indicators_and_weights( # an input matrix without alternatives & unuseful columns
+                self.input_matrix_no_alternatives,
+                self.robustness_indicators,
+                self.robustness_weights,
+                self.robustness_single_weights,
+                self.polarity, self.num_runs,
+                weights, self.marginal_distributions)
+            self.weights = weights.copy()
+
+        validate_configuration(
+            input_matrix=self.input_matrix,
+            polarity=self.polarity,
+            weights=self.weights,
```

Comment:
I think same applies to all the parameters. You have already used them in the code but validating them now.

Reply:
The two previous steps are necessary to `validate_configuration`.

Example: if the input matrix has a column with no information, this column is dropped in `check_input_matrix`. The weights and polarities are consequently shortened in `process_indicators_and_weights`. Only then can we validate the entire configuration settings, for example, estimate the number of indicators in the case of robustness analysis in validate_configuration.

This logic and all the controls could be rewritten more organically and clearly. I leave this improvement to a later version of the library.